uvicorn/protocols/http/httptools_impl.py

```diff
@@ -429,11 +429,19 @@ def __init__(
             headers = self.default_headers + list(message.get("headers", []))
 
             if self.access_log:
+                path_and_query_string = (
```

Comment:
I'm not delighted at this code being duplicated in both `httptools_impl.py` and `h11_impl.py` - maybe this should be a helper function in `uvicorn.protocols.utils`?

Reply:
Fixed in https://github.com/encode/uvicorn/pull/384/commits/75c39c2ef41327e4569324498efaf76375c9a3c3 - I think this is better.